python/cuml/solvers/qn.pyx

```diff
@@ -102,7 +102,103 @@ cdef extern from "glm/glm.hpp" namespace "ML::GLM":
 
 class QN(Base):
     """
-    Quasi-Newton Goodness :)
+    Quasi-Newton methods are methods used to either find zeroes or local maxima
+    and minima of functions, and used by this class to optimize a cost
+    function.
+
+    Two algorithms are implemented underneath cuML's QN class, and which one
+    is executed depends on the following rule:
+
+    - Orthant-Wise Limited Memory Quasi-Newton (OWL-QN) if there is l1
+    regularization
+    - Limited Memory BFGS (L-BFGS) otherwise.
+
+    cuML's QN class can take array-like objects, either in host as
+    NumPy arrays or in device (as Numba or __cuda_array_interface__ compliant).
+
+    Examples
+    ---------
+    .. code-block:: python
+
+        import cudf
+        import numpy as np
+
+        # Both import methods supported
+        # from cuml import QN
+        from cuml.solvers import QN
+
+        X = cudf.DataFrame()
+        X['col1'] = np.array([1,1,2,2], dtype = np.float32)
+        X['col2'] = np.array([1,2,2,3], dtype = np.float32)
+        y = cudf.Series( np.array([0.0, 0.0, 1.0, 1.0], dtype = np.float32) )
+
+        solver = QN()
+        solver.fit(X,y)
+
+        # Note: for now, the coefficients also include the intercept in the
+        # last position if fit_intercept=True
+        print("Coefficients:")
+        print(solver.coef_.copy_to_host())
+        print("intercept:")
+        print(solver.intercept_.copy_to_host())
+
+        X_new = cudf.DataFrame()
+        X_new['col1'] = np.array([1,5], dtype = np.float32)
+        X_new['col2'] = np.array([2,5], dtype = np.float32)
+
+        preds = solver.predict(X_new)
+
```

Comment:
print("Predictions:")

This doesn't correspond exactly to the output otherwise.